internal/http_helper.go

```diff
@@ -7,72 +7,65 @@ import (
 	"strings"
 	"time"
 
-	metrics "github.com/armon/go-metrics"
 	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/cenkalti/backoff"
 	docker "github.com/fsouza/go-dockerclient"
 	"github.com/gorilla/mux"
-	log "github.com/sirupsen/logrus"
-)
-
-const (
-	retryCount = 5
-	retrySleep = 5 * time.Millisecond
 )
 
 func remoteIP(addr string) string {
 	return strings.Split(addr, ":")[0]
 }
 
-func findContainerRoleByAddress(addr string, labels []metrics.Label) (*iam.Role, []metrics.Label, error) {
+func findContainerRoleByAddress(addr string, request *Request) (*iam.Role, error) {
 	var container *docker.Container
 
 	// retry finding the Docker container since sometimes Docker doesn't actually list the container until its been
 	// running for a while. This is a really simple and basic retry policy
-	var err error
 	remoteIP := remoteIP(addr)
-	for i := 1; i <= retryCount; i++ {
-		container, labels, err = findDockerContainer(remoteIP, labels)
-		// if we got no errors, just break the loop and keep moving forward
-		if err == nil {
-			break
-		}
-
-		// if we got an error, log that and take a quick nap
-		logWithLabels(labels).Errorf("Could not find Docker container with remote IP %s (retry %d out of %d)", remoteIP, i, retryCount)
-		time.Sleep(retrySleep)
+
+	b := backoff.NewExponentialBackOff()
+	b.MaxElapsedTime = 5 * time.Second
```

Comment:
👏

Reply:
You should also set `b.InitialInterval = 5 * time.Millisecond` to preserve the previous behavior